units/initrd-parse-etc.service.in

```diff
@@ -8,17 +8,25 @@
 #  (at your option) any later version.
 
 [Unit]
-Description=Reload Configuration from the Real Root
+Description=Mountpoints Configured in the Real Root
+AssertPathExists=/etc/initrd-release
+
 DefaultDependencies=no
 Requires=initrd-root-fs.target
 After=initrd-root-fs.target
+
 OnFailure=emergency.target
 OnFailureJobMode=replace-irreversibly
-AssertPathExists=/etc/initrd-release
 
 [Service]
 Type=oneshot
-ExecStartPre=-systemctl daemon-reload
-# we have to retrigger initrd-fs.target after daemon-reload
-ExecStart=-systemctl --no-block start initrd-fs.target
+
+ExecStart={{SYSTEM_GENERATOR_DIR}}/systemd-fstab-generator --sysroot-only \
+  /run/systemd/generator/ /run/systemd/generator.early/ /run/systemd/generator.late/
```

Comment:
hmm, what? Running a generator from a service? this is supposed to add more units to the generator dirs?

that's not safe, if a unit the generator generates was already referenced before it might be loaded already and then would not be reloaded once the generator is finished.

I mean, I really dislike running generators out of generator concepts.

Reply:
Please see the commit message. It explains this in some detail. (My reasoning may be wrong, but please at least read the description.)